### PR TITLE
Bugfix/iat 2151

### PR DESCRIPTION
**Pull request by jeffjohnson9046**
[IAT-2151](https://jira.fairwaytech.com/browse/IAT-2151):
* Add endpoint that allows the caller to abandon a job (stop the job if it's in a running state then mark it as abandoned).
* Additional feedback in `JobExecutionResponse`

### CODE REVIEW

File: src/main/java/org/opentestsystem/ap/imrt/iis/service/impl/SpringBatchJobManagementServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    public SpringBatchJobManagementServiceImpl(final JobExplorer jobExplorer, final JobOperator jobOperator) {
+        this.jobExplorer = jobExplorer;
+        this.jobOperator = jobOperator;
+    }
+
+    @Override
+    public JobExecutionResponse abandon(final long jobInstanceId) {
+        final JobInstance jobInstance = Optional.ofNullable(jobExplorer.getJobInstance(jobInstanceId))
+                .orElseThrow(() -> new IllegalStateException(String.format("Could not find job instance for instance id %d", jobInstanceId)));
```
Comment (jeffjohnson9046) [inline src/main/java/org/opentestsystem/ap/imrt/iis/service/impl/SpringBatchJobManagementServiceImpl.java]: I went back & forth on this a bit; I suppose I could also throw one of the `JobAbandonmentException`s I created here.  I figured if we can't find a job for the instance id, we're in a bad way so an `IllegalStateException` is a better fit.

File: src/main/java/org/opentestsystem/ap/imrt/iis/service/impl/SpringBatchJobManagementServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+        final Set<JobExecution> runningJobExecutions = jobExplorer.findRunningJobExecutions(jobInstance.getJobName());
+
+        // If more than one running job execution is found, throw exception.  Based on how the the jobs are configured,
+        // this should never happen
+        if (runningJobExecutions.size() > 1) {
```
Comment (jeffjohnson9046) [inline src/main/java/org/opentestsystem/ap/imrt/iis/service/impl/SpringBatchJobManagementServiceImpl.java]: I went back & forth on this too; I don't see a way for the Spring Batch to get into this state aside from a code/programming bug.

File: src/main/java/org/opentestsystem/ap/imrt/iis/service/SpringBatchJobManagementService.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package org.opentestsystem.ap.imrt.iis.service;
+
+import org.opentestsystem.ap.imrt.iis.model.JobExecutionResponse;
+
+/**
+ * A service for interacting with Spring Batch jobs.
+ */
+public interface SpringBatchJobManagementService {
```
Comment (jeffjohnson9046) [inline src/main/java/org/opentestsystem/ap/imrt/iis/service/SpringBatchJobManagementService.java]: Long term, I think I'd like the controller to call methods on this interface to start the `sync` and/or `migrate` jobs.

File: src/main/java/org/opentestsystem/ap/imrt/iis/service/impl/SpringBatchJobManagementServiceImpl.java (hunk 1 above)
Comment (millerts) [reply src/main/java/org/opentestsystem/ap/imrt/iis/service/impl/SpringBatchJobManagementServiceImpl.java]: Isn't this something a user passes in?  If so isn't a NotFoundException more appropriate?  Or something that would trigger a 404?

**Pull request merged**